include/ImNodeFlow.h

```diff
@@ -310,6 +310,30 @@ namespace ImFlow
         template<typename T, typename... Params>
         std::shared_ptr<T> addNode(const ImVec2& pos, Params&&... args);
 
+        /**
+         * @brief <BR>Helper struct for creating a node struct from a lambda.
+         * @tparam L the type of the lambda
+         * @tparam B always BaseNode, tparam because BaseNode is incomplete here
+         */
+        template <typename L, typename B = BaseNode>
+        struct NodeWrapper : public B, public L
+        {
+            NodeWrapper(L&& l): BaseNode(), L(std::forward<L>(l)) {}
+            void draw() { L::operator()(this); }
+        };
+
+        /**
+         * @brief <BR>Add a node whos operation can be defined within a lambda.
+         * @tparam L the type of the lambda
+         * @param lambda the lambda that defines the nodes operation
+         * @param pos the position at which to place the node
+         */
+        template<typename L>
+        std::shared_ptr<NodeWrapper<L>> addLambda(L&& lambda, const ImVec2& pos)
```

Comment:
I prefer addLambdaNode
Also, did you test the nodes behavior with the various interactions and its deletion?

Reply:
UI? I'm afraid I don't understand what you meant

Reply:
A user interface element for registering nodes to add? For example right clicking on the grid could have a way to add nodes of specific types that had previously been registered.

My example also no longer worked from some update made so it was updated to the below example, which also doesn't work.
```cpp
INF.addLambda([](ImFlow::BaseNode* self) {
  const auto val = self->showIN<double>("INPUT", 0.0, ImFlow::ConnectionFilter::SameType());
  self->showOUT<double>("OUTPUT", [val](){
    return val;
  });
}, {0, 0});
```

I have traced this to some unexpected behaviour of `showOUT` as I can also reproduce this problem with a struct.
```cpp
struct Test : ImFlow::BaseNode {
  void draw(){
    double val = this->showIN<double>("INPUT", 0.0, ImFlow::ConnectionFilter::SameType());
    this->showOUT<double>("OUTPUT", [=](){
      return val;
    });
  }
};
```
You would expect in this example that the output pin would always mirror the current value of the input pin. However, in both these examples the program only outputs the value `val` has the first time, which is `0`, because the default takes place before a connection can occur. This is because when the pin is found in the dynamic pin list after not being found the first time resulting in creation, in all subsequent calls to `showOUT` the `std::function<T()>` doesn't get updated. To get that working a patch here to update the behaviour is required https://github.com/Fattorino/ImNodeFlow/blob/master/src/ImNodeFlow.inl#L117 adding a `static_cast<OutPin<T>*>(m_dynamicOuts.back().second.get())->behaviour(std::move(behaviour));` before the `return`.

Reply:
Back to testing the adding and removal it seems to work from testing with a button to add new nodes and deleting the nodes that had been created previously.

Reply:
I also noticed my example for the docs no longer works from one of the updates when this stuff was under change so it was updated to the below example, which also doesn't work.
```cpp
INF.addLambda([](ImFlow::BaseNode* self) {
  const auto val = self->showIN<double>("INPUT", 0.0, ImFlow::ConnectionFilter::SameType());
  self->showOUT<double>("OUTPUT", [val](){
    return val;
  });
}, {0, 0});
```

I have traced this to some unexpected behaviour of `showOUT` as I can also reproduce this problem with a struct.
```cpp
struct Test : ImFlow::BaseNode {
  void draw(){
    double val = this->showIN<double>("INPUT", 0.0, ImFlow::ConnectionFilter::SameType());
    this->showOUT<double>("OUTPUT", [=](){
      return val;
    });
  }
};
```
You would expect in this example that the output pin would always mirror the current value of the input pin. However, in both these examples the program only outputs the value `val` has the first time, which is `0`, because the default takes place before a connection can occur. This is because when the pin is found in the dynamic pin list after not being found the first time resulting in creation, in all subsequent calls to `showOUT` the `std::function<T()>` doesn't get updated. To get that working a patch here to update the behaviour is required https://github.com/Fattorino/ImNodeFlow/blob/master/src/ImNodeFlow.inl#L117 adding a `static_cast<OutPin<T>*>(m_dynamicOuts.back().second.get())->behaviour(std::move(behaviour));` before the `return`.

Reply:
Changing the template signature could be interesting, if you'd like to give it a try I'm curious to see how it turns out

Reply:
> Tested with statically adding. I can give adding/deleting the nodes a try, but shouldn't be any different than the struct approach. In the vein of adding/removing it might be interesting to build some default UI for registering nodes under a creation button/menu.
> 
> I'll test later today.

I was working on something similar i think
https://gist.github.com/Riztazz/3967644d2843492bce541f25c4fdf26c

and i use it as such;
```
            using ViewportNodes2 = ViewportNodes<SimpleSum, SomeEntryNodeInputs, SomeEntryNodeNoInputs, CommentNode, NodeWithJustInput>;
            graph.rightClickPopUpContent( []( ImFlow::BaseNode * node )
            {
                if ( !node )
                {
                    for ( uint32_t nodeIndex = 0; nodeIndex < ViewportNodes2::nodeMetas.size(); ++nodeIndex )
                    {
                        if ( Button( ViewportNodes2::GetNodeName( nodeIndex ).data() ) )
                        {
                            ViewportNodes2::CreateNode( nodeIndex, graph );
                            CloseCurrentPopup();
                        }
                    }
                }
            } );
```

Reply:
Honestly, I hadn't thought about that, but unfortunately, I don't have the time to investigate or develop anything. But I'll try my best to review and merge PR to keep the project alive.

Reply:
In my opinion, it shouldn't be part of library itself anyway, there is too many ways to go about it and it makes the codebase easier to get into.
std::monostate way of deprecation seems to work in case anyone needs it;
![image](https://github.com/user-attachments/assets/55263b97-e7c1-4c3f-b7fc-e2cd12ed4081)